doc/reference/multisig.md

```diff
@@ -8,57 +8,230 @@ The multi-signature scripts are written as JSON objects. We have three types of
 
 ```json
 {
-    "atLeast": {
-        "list": [
-            "2f3d4cf10d0471a1db9f2d2907de867968c27bca6272f062cd1c2413",
-            "f856c0c5839bab22673747d53f1ae9eed84afafb085f086e8e988614",
-            "b275b08c999097247f7c17e77007c7010cd19f20cc086ad99d398538",
-            "686024aecb5884d73a11b9ae4e63931112ba737e878d74638b78513a"
-        ],
-        "required": 2
-    }
+    "scripts": [
+        {
+            "paymentKeyHash": "2f3d4cf10d0471a1db9f2d2907de867968c27bca6272f062cd1c2413",
+            "type": "requireSignature"
+        },
+        {
+            "paymentKeyHash": "f856c0c5839bab22673747d53f1ae9eed84afafb085f086e8e988614",
+            "type": "requireSignature"
+        },
+        {
+            "paymentKeyHash": "b275b08c999097247f7c17e77007c7010cd19f20cc086ad99d398538",
+            "type": "requireSignature"
+        },
+        {
+            "paymentKeyHash": "686024aecb5884d73a11b9ae4e63931112ba737e878d74638b78513a",
+            "type": "requireSignature"
+        }
+    ],
+    "required": 2,
+    "type": "atLeast"
 }
 ```
+### Example of using multi-signature scripts
+
+Let's walk through how one would use a multi-signature script. This is a multistep step process, involving the creation of a multi-signature address, sending ADA to that address followed by gathering the required witnesses in order to spend the ADA from the multi-signature address. We will demonstrate this with an `all` script.
+
+#### Sending ada to a script address
+
+#### Step 1 - Create multi-signature script
+
+You will first need to generate the keys that you require witnessing from via the `cardano-cli shelley  address key-gen ` command. Then you construct the multi-signature script as follows:
+
+```
+cardano-cli -- shelley transaction build-multisig
+  --all
+  --payment-verification-key-file payVerKey1
+  --payment-verification-key-file payVerKey2
+  --payment-verification-key-file payVerKey3
+  --out-file allMultiSigScript
+```
+This will output a JSON file with the `all` format described above.
+
+
+#### Step 2 - Create multi-signature address
+We need a special multi-signature address to use with our multi-signature script. Please note the network magic must match with your network's magic (or if its mainnet use the `--mainnet` flag instead). This can be constructed as follows:
+```
+cardano-cli shelley address build-script
+  --script-file allMultiSigScript
+  --testnet-magic 42
+  --out-file script.addr
+```
+
+#### Step 3 - Construct and submit a tx to the multi-signature address
+We then need to construct and submit a tx to send ADA to the multi-signature address.
+
+Construct the tx body:
+```
+cardano-cli shelley transaction build-raw
+    --ttl 1000
+    --fee 0
+    --tx-in utxoinput
+    --tx-out $(cat script.addr)+$amount
```

Comment:
Are we certain that `script.addr` will never contain extraneous whitespace that could screw this expansion up?

What does a typical contents look like?

Reply:
Typical contents: `addr_test1wrffpllpuslgpva4dz3r5r6nyw6mjt6mngemasl53mkk2hs2qf34q`